tests/test_dataset.py

```diff
@@ -0,0 +1,520 @@
+import filecmp
+
+import numpy as np
+from shutil import rmtree, copytree
+
+from wkcuber.api.Dataset import WKDataset, TiffDataset
+from os import path, mkdir
+
+from wkcuber.api.Properties import Properties, TiffProperties
+
+
+def delete_dir(relative_path):
+    if path.exists(relative_path) and path.isdir(relative_path):
+        rmtree(relative_path)
+
+
+def test_create_WKDataset_with_layer_and_mag():
+    delete_dir("../testoutput/WKDataset")
+
+    ds = WKDataset.create("../testoutput/WKDataset", scale=(1, 1, 1))
+    ds.add_layer("color", "color")
+
+    ds.get_layer("color").add_mag("1")
+    ds.get_layer("color").add_mag("2-2-1")
+
+    assert path.exists("../testoutput/WKDataset/color/1")
+    assert path.exists("../testoutput/WKDataset/color/2-2-1")
+
+    assert len(ds.properties.data_layers) == 1
+    assert len(ds.properties.data_layers["color"].wkw_resolutions) == 2
+
+
+def test_create_WKDataset_with_explicit_header_fields():
+    delete_dir("../testoutput/WKDataset_advanced")
+
+    ds = WKDataset.create("../testoutput/WKDataset_advanced", scale=(1, 1, 1))
+    ds.add_layer("color", "color", dtype=np.uint16, num_channels=3)
+
+    ds.get_layer("color").add_mag("1", block_len=64, file_len=64)
+    ds.get_layer("color").add_mag("2-2-1")
+
+    assert path.exists("../testoutput/WKDataset_advanced/color/1")
+    assert path.exists("../testoutput/WKDataset_advanced/color/2-2-1")
+
+    assert len(ds.properties.data_layers) == 1
+    assert len(ds.properties.data_layers["color"].wkw_resolutions) == 2
+
+    assert ds.properties.data_layers["color"].element_class == np.dtype(np.uint16)
+    assert (
+        ds.properties.data_layers["color"].wkw_resolutions[0].cube_length == 64 * 64
+    )  # mag "1"
+    assert (
+        ds.properties.data_layers["color"].wkw_resolutions[1].cube_length == 32 * 32
+    )  # mag "2-2-1" (defaults are used)
+
+
+def test_create_TiffDataset_with_layer_and_mag():
+    # This test would be the same for WKDataset
+
+    delete_dir("../testoutput/TiffDataset")
+
+    ds = WKDataset.create("../testoutput/TiffDataset", scale=(1, 1, 1))
+    ds.add_layer("color", "color")
+
+    ds.get_layer("color").add_mag("1")
+    ds.get_layer("color").add_mag("2-2-1")
+
+    assert path.exists("../testoutput/TiffDataset/color/1")
+    assert path.exists("../testoutput/TiffDataset/color/2-2-1")
+
+    assert len(ds.properties.data_layers) == 1
+    assert len(ds.properties.data_layers["color"].wkw_resolutions) == 2
+
+
+def test_open_wk_dataset():
+    ds = WKDataset.open("../testdata/simple_wk_dataset")
```

Comment:
This path (and the ones below) use snake_case and the above paths used PascalCase. Could you make this consistent?

Reply:
Since the `open` call doesn't need a corresponding `close` call, what do you think about changing the interface to a simple constructor? Like `WkDataset("...")`? I'd also support @normanrz suggestion for allowing a get-or-create behavior via an additional argument.